client/src/pages/Join/Join.jsx

```diff
@@ -43,12 +67,21 @@ const Join = () => {
       </div>
       <div className="join-left-container">
         <h1 className="join-header">Start chatting!</h1>
+        {error && <p>{error}</p>}
         <input autoComplete="off" placeholder="Username" id="username" className="join-input" type="text" onChange={(event) => setFormUsername(event.target.value)} value={formName} />
-        <input autoComplete="off" placeholder="Room" id="room" className="join-input" type="text" onChange={(event) => setFormRoom(event.target.value)} />
-        <Link className="join-button" onClick={(event) => join(event)} to="/chat">Sign in</Link>
+        <input autoComplete="off" placeholder="Room" id="room" className="join-input" type="text" onChange={(event) => setFormRoom(event.target.value)}  value={formRoom} />
+        <button type="button" className="join-button" onClick={(event) => join(event)}>Sign in</button>
       </div>
+      {showModal && <div className="join-overlay" />}
+      {showModal && <Modal />}
```

Comment:
Because L75 is part of the `showModal` conditional, could you add it inside of the `<Modal>` component?